proposed/2022/duplicate-nuget-item-error-handling.md

```diff
@@ -0,0 +1,142 @@
+# Duplicate NuGet item error handling
+
+- Author Name: [nkolev92](https://github.com/nkolev92)
+- Start Date 2021-04-01
+- GitHub Issue - Duplicate PackageReference [9864](https://github.com/NuGet/Home/issues/9864), CPVM duplicate erroring [9467](https://github.com/NuGet/Home/issues/9467)
+- GitHub PR [4484](https://github.com/NuGet/NuGet.Client/pull/4484)
+
+## Summary
+
+It's not uncommon that PackageReference (or PackageVersion, PackageDownload etc) items are included through secondary files and/or are manually edited, which can sometimes lead to multiple items with the same name being included.
+In some scenarios, this can cause restore inconsistencies. We will to detect these scenarios and raise a warning. NoWarn and TreatWarningsAsErrors are also respected.
+
+## Motivation
+
+Take the following project:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="NuGet.Protocol" Version="5.5.0" />
+    <PackageReference Include="NuGet.Protocol" Version="5.6.0" />
```

Comment:
Will this new duplicate check consider when you have:

- PackageReference declarations with package ID's only differing by case, e.g., pkg1 and Pkg1?
- PackageReference declarations with same id but different versions?

Reply:
duplication is case insensitive.

The version being different/same is not relevant.